// mongodb-stitch-react-native-core/dist/cjs/core/internal/net/RNFetchStreamTransport.js
"use strict";
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var mongodb_stitch_core_sdk_1 = require("mongodb-stitch-core-sdk");
var EventSourceEventStream_1 = __importDefault(require("./EventSourceEventStream"));
var eventsource_1 = __importDefault(require("eventsource"));
var cross_fetch_1 = __importDefault(require("cross-fetch"));
var RNFetchStreamTransport = (function () {
    function RNFetchStreamTransport() {
    }
    RNFetchStreamTransport.prototype.roundTrip = function (request) {
        var responsePromise = cross_fetch_1.default(request.url, {
            body: request.body,
            headers: request.headers,
            method: request.method,
            mode: 'cors'
        });
        var responseTextPromise = responsePromise.then(function (response) {
            return response.text();
        });
        return Promise.all([responsePromise, responseTextPromise]).then(function (values) {
            var response = values[0];
            var body = values[1];
            var headers = {};
            response.headers.forEach(function (value, key) {
                headers[key] = value;
            });
            return new mongodb_stitch_core_sdk_1.Response(headers, response.status, body);
        });
    };
    RNFetchStreamTransport.prototype.stream = function (request, open, retryRequest) {
        if (open === void 0) { open = true; }
        var headers = __assign({}, request.headers);
        headers[mongodb_stitch_core_sdk_1.Headers.ACCEPT] = mongodb_stitch_core_sdk_1.ContentTypes.TEXT_EVENT_STREAM;
        headers[mongodb_stitch_core_sdk_1.Headers.CONTENT_TYPE] = mongodb_stitch_core_sdk_1.ContentTypes.TEXT_EVENT_STREAM;
        return cross_fetch_1.default(request.url + "&stitch_validate=true", {
            body: request.body,
            headers: headers,
            method: request.method,
            mode: 'cors'
        }).then(function (response) {
            var headers = {};
            response.headers.forEach(function (value, key) {
                headers[key] = value;
            });
            if (response.status < 200 || response.status >= 300) {
                return response.text()
                    .then(function (body) { return mongodb_stitch_core_sdk_1.handleRequestError(new mongodb_stitch_core_sdk_1.Response(headers, response.status, body)); });
            }
            return new Promise(function (resolve, reject) {
                new EventSourceEventStream_1.default(new eventsource_1.default(request.url), function (stream) { return resolve(stream); }, function (error) { return reject(error); }, retryRequest ?
                    function () { return retryRequest().then(function (es) { return es; }); }
                    : undefined);
            });
        });
    };
    return RNFetchStreamTransport;
}());
exports.default = RNFetchStreamTransport;
