// mongodb-stitch-react-native-core/dist/cjs/core/internal/StitchAppClientImpl.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var mongodb_stitch_core_sdk_1 = require("mongodb-stitch-core-sdk");
var StitchServiceClientImpl_1 = __importDefault(require("../../services/internal/StitchServiceClientImpl"));
var StitchAuthImpl_1 = __importDefault(require("../auth/internal/StitchAuthImpl"));
var StitchAppClientImpl = (function () {
    function StitchAppClientImpl(clientAppId, config) {
        this.info = new mongodb_stitch_core_sdk_1.StitchAppClientInfo(clientAppId, config.dataDirectory, config.localAppName, config.localAppVersion);
        this.routes = new mongodb_stitch_core_sdk_1.StitchAppRoutes(this.info.clientAppId);
        var requestClient = new mongodb_stitch_core_sdk_1.StitchAppRequestClient(clientAppId, config.baseUrl, config.transport);
        this.auth = new StitchAuthImpl_1.default(requestClient, this.routes.authRoutes, config.storage, this.info);
        this.coreClient = new mongodb_stitch_core_sdk_1.CoreStitchAppClient(this.auth, this.routes);
    }
    StitchAppClientImpl.prototype.getServiceClient = function (factory, serviceName) {
        if (isServiceClientFactory(factory)) {
            return factory.getClient(new mongodb_stitch_core_sdk_1.CoreStitchServiceClientImpl(this.auth, this.routes.serviceRoutes, ""), this.info);
        }
        else {
            return factory.getNamedClient(new mongodb_stitch_core_sdk_1.CoreStitchServiceClientImpl(this.auth, this.routes.serviceRoutes, serviceName), this.info);
        }
    };
    StitchAppClientImpl.prototype.getGeneralServiceClient = function (serviceName) {
        return new StitchServiceClientImpl_1.default(new mongodb_stitch_core_sdk_1.CoreStitchServiceClientImpl(this.auth, this.routes.serviceRoutes, serviceName));
    };
    StitchAppClientImpl.prototype.callFunction = function (name, args) {
        return this.coreClient.callFunction(name, args);
    };
    StitchAppClientImpl.prototype.close = function () {
        this.auth.close();
    };
    return StitchAppClientImpl;
}());
exports.default = StitchAppClientImpl;
function isServiceClientFactory(factory) {
    return factory.getClient !== undefined;
}
