// mongodb-stitch-core-sdk/dist/cjs/auth/providers/custom/CustomCredential.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var ProviderCapabilities_1 = __importDefault(require("../../ProviderCapabilities"));
var CustomAuthProvider_1 = __importDefault(require("./CustomAuthProvider"));
var Fields;
(function (Fields) {
    Fields["TOKEN"] = "token";
})(Fields || (Fields = {}));
var CustomCredential = (function () {
    function CustomCredential(token, providerName) {
        var _a;
        if (providerName === void 0) { providerName = CustomAuthProvider_1.default.DEFAULT_NAME; }
        this.providerType = CustomAuthProvider_1.default.TYPE;
        this.providerCapabilities = new ProviderCapabilities_1.default(false);
        this.providerName = providerName;
        this.token = token;
        this.material = (_a = {}, _a[Fields.TOKEN] = this.token, _a);
    }
    return CustomCredential;
}());
exports.default = CustomCredential;
