// mongodb-stitch-react-native-core/dist/esm/core/auth/internal/StitchAuthImpl.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { AuthEventKind, CoreStitchAuth, DeviceFields, StitchClientError, StitchClientErrorCode } from "mongodb-stitch-core-sdk";
import version from "../../internal/common/Version";
import StitchUserFactoryImpl from "./StitchUserFactoryImpl";
var StitchAuthImpl = (function (_super) {
    __extends(StitchAuthImpl, _super);
    function StitchAuthImpl(requestClient, authRoutes, authStorage, appInfo) {
        var _this = _super.call(this, requestClient, authRoutes, authStorage) || this;
        _this.appInfo = appInfo;
        _this.listeners = new Set();
        _this.synchronousListeners = new Set();
        return _this;
    }
    Object.defineProperty(StitchAuthImpl.prototype, "userFactory", {
        get: function () {
            return new StitchUserFactoryImpl(this);
        },
        enumerable: true,
        configurable: true
    });
    StitchAuthImpl.prototype.getProviderClient = function (factory, providerName) {
        if (isAuthProviderClientFactory(factory)) {
            return factory.getClient(this, this.requestClient, this.authRoutes);
        }
        else {
            return factory.getNamedClient(providerName, this.requestClient, this.authRoutes);
        }
    };
    StitchAuthImpl.prototype.loginWithCredential = function (credential) {
        return _super.prototype.loginWithCredentialInternal.call(this, credential);
    };
    StitchAuthImpl.prototype.linkWithCredential = function (user, credential) {
        return _super.prototype.linkUserWithCredentialInternal.call(this, user, credential);
    };
    StitchAuthImpl.prototype.logout = function () {
        if (arguments.length > 0) {
            return Promise.reject(new StitchClientError(StitchClientErrorCode.UnexpectedArguments));
        }
        return _super.prototype.logoutInternal.call(this);
    };
    StitchAuthImpl.prototype.logoutUserWithId = function (userId) {
        return _super.prototype.logoutUserWithIdInternal.call(this, userId);
    };
    StitchAuthImpl.prototype.removeUser = function () {
        if (arguments.length > 0) {
            return Promise.reject(new StitchClientError(StitchClientErrorCode.UnexpectedArguments));
        }
        return _super.prototype.removeUserInternal.call(this);
    };
    StitchAuthImpl.prototype.removeUserWithId = function (userId) {
        return _super.prototype.removeUserWithIdInternal.call(this, userId);
    };
    Object.defineProperty(StitchAuthImpl.prototype, "deviceInfo", {
        get: function () {
            var info = {};
            if (this.hasDeviceId) {
                info[DeviceFields.DEVICE_ID] = this.deviceId;
            }
            if (this.appInfo.localAppName !== undefined) {
                info[DeviceFields.APP_ID] = this.appInfo.localAppName;
            }
            if (this.appInfo.localAppVersion !== undefined) {
                info[DeviceFields.APP_VERSION] = this.appInfo.localAppVersion;
            }
            info[DeviceFields.PLATFORM] = "js-react-native";
            info[DeviceFields.PLATFORM_VERSION] = process.version;
            info[DeviceFields.SDK_VERSION] = version;
            return info;
        },
        enumerable: true,
        configurable: true
    });
    StitchAuthImpl.prototype.addAuthListener = function (listener) {
        this.listeners.add(listener);
        this.onAuthEvent(listener);
        this.dispatchAuthEvent({
            kind: AuthEventKind.ListenerRegistered,
        });
    };
    StitchAuthImpl.prototype.addSynchronousAuthListener = function (listener) {
        this.listeners.add(listener);
        this.onAuthEvent(listener);
        this.dispatchAuthEvent({
            kind: AuthEventKind.ListenerRegistered,
        });
    };
    StitchAuthImpl.prototype.removeAuthListener = function (listener) {
        this.listeners.delete(listener);
    };
    StitchAuthImpl.prototype.onAuthEvent = function (listener) {
        var _this = this;
        if (listener) {
            var _1 = new Promise(function (resolve) {
                if (listener.onAuthEvent) {
                    listener.onAuthEvent(_this);
                }
                resolve(undefined);
            });
        }
        else {
            this.listeners.forEach(function (one) {
                _this.onAuthEvent(one);
            });
        }
    };
    StitchAuthImpl.prototype.dispatchAuthEvent = function (event) {
        var _this = this;
        switch (event.kind) {
            case AuthEventKind.ActiveUserChanged:
                this.dispatchBlockToListeners(function (listener) {
                    if (listener.onActiveUserChanged) {
                        listener.onActiveUserChanged(_this, event.currentActiveUser, event.previousActiveUser);
                    }
                });
                break;
            case AuthEventKind.ListenerRegistered:
                this.dispatchBlockToListeners(function (listener) {
                    if (listener.onListenerRegistered) {
                        listener.onListenerRegistered(_this);
                    }
                });
                break;
            case AuthEventKind.UserAdded:
                this.dispatchBlockToListeners(function (listener) {
                    if (listener.onUserAdded) {
                        listener.onUserAdded(_this, event.addedUser);
                    }
                });
                break;
            case AuthEventKind.UserLinked:
                this.dispatchBlockToListeners(function (listener) {
                    if (listener.onUserLinked) {
                        listener.onUserLinked(_this, event.linkedUser);
                    }
                });
                break;
            case AuthEventKind.UserLoggedIn:
                this.dispatchBlockToListeners(function (listener) {
                    if (listener.onUserLoggedIn) {
                        listener.onUserLoggedIn(_this, event.loggedInUser);
                    }
                });
                break;
            case AuthEventKind.UserLoggedOut:
                this.dispatchBlockToListeners(function (listener) {
                    if (listener.onUserLoggedOut) {
                        listener.onUserLoggedOut(_this, event.loggedOutUser);
                    }
                });
                break;
            case AuthEventKind.UserRemoved:
                this.dispatchBlockToListeners(function (listener) {
                    if (listener.onUserRemoved) {
                        listener.onUserRemoved(_this, event.removedUser);
                    }
                });
                break;
            default:
                return this.assertNever(event);
        }
    };
    StitchAuthImpl.prototype.assertNever = function (x) {
        throw new Error("unexpected object: " + x);
    };
    StitchAuthImpl.prototype.dispatchBlockToListeners = function (block) {
        this.synchronousListeners.forEach(block);
        this.listeners.forEach(function (listener) {
            var _ = new Promise(function (resolve) {
                block(listener);
                resolve(undefined);
            });
        });
    };
    return StitchAuthImpl;
}(CoreStitchAuth));
export default StitchAuthImpl;
function isAuthProviderClientFactory(factory) {
    return (factory.getClient !== undefined);
}
