// mongodb-stitch-core-sdk/dist/esm/index.js
import BSON from "bson";
import { AuthEventKind } from "./auth/internal/AuthEvent";
import AuthInfo from "./auth/internal/AuthInfo";
import CoreStitchAuth from "./auth/internal/CoreStitchAuth";
import CoreStitchUserImpl from "./auth/internal/CoreStitchUserImpl";
import DeviceFields from "./auth/internal/DeviceFields";
import ApiStitchUserIdentity from "./auth/internal/models/ApiStitchUserIdentity";
import StitchUserProfileImpl from "./auth/internal/StitchUserProfileImpl";
import AnonymousAuthProvider from "./auth/providers/anonymous/AnonymousAuthProvider";
import AnonymousCredential from "./auth/providers/anonymous/AnonymousCredential";
import CustomAuthProvider from "./auth/providers/custom/CustomAuthProvider";
import CustomCredential from "./auth/providers/custom/CustomCredential";
import FacebookAuthProvider from "./auth/providers/facebook/FacebookAuthProvider";
import FacebookCredential from "./auth/providers/facebook/FacebookCredential";
import GoogleAuthProvider from "./auth/providers/google/GoogleAuthProvider";
import GoogleCredential from "./auth/providers/google/GoogleCredential";
import StitchAuthResponseCredential from "./auth/providers/internal/StitchAuthResponseCredential";
import ServerApiKeyAuthProvider from "./auth/providers/serverapikey/ServerApiKeyAuthProvider";
import ServerApiKeyCredential from "./auth/providers/serverapikey/ServerApiKeyCredential";
import CoreUserApiKeyAuthProviderClient from "./auth/providers/userapikey/CoreUserApiKeyAuthProviderClient";
import UserApiKey from "./auth/providers/userapikey/models/UserApiKey";
import UserApiKeyAuthProvider from "./auth/providers/userapikey/UserApiKeyAuthProvider";
import UserApiKeyCredential from "./auth/providers/userapikey/UserApiKeyCredential";
import CoreUserPassAuthProviderClient from "./auth/providers/userpass/CoreUserPasswordAuthProviderClient";
import UserPasswordAuthProvider from "./auth/providers/userpass/UserPasswordAuthProvider";
import UserPasswordCredential from "./auth/providers/userpass/UserPasswordCredential";
import StitchUserIdentity from "./auth/StitchUserIdentity";
import UserType from "./auth/UserType";
import Assertions from "./internal/common/Assertions";
import { base64Decode, base64Encode, utf8Slice } from "./internal/common/Base64";
import { handleRequestError } from "./internal/common/StitchErrorUtils";
import { MemoryStorage } from "./internal/common/Storage";
import CoreStitchAppClient from "./internal/CoreStitchAppClient";
import BaseEventStream from "./internal/net/BaseEventStream";
import { BasicRequest } from "./internal/net/BasicRequest";
import ContentTypes from "./internal/net/ContentTypes";
import Event from "./internal/net/Event";
import Headers from "./internal/net/Headers";
import Method from "./internal/net/Method";
import Response from "./internal/net/Response";
import StitchAppAuthRoutes from "./internal/net/StitchAppAuthRoutes";
import StitchAppRequestClient from "./internal/net/StitchAppRequestClient";
import StitchAppRoutes from "./internal/net/StitchAppRoutes";
import { StitchAuthRequest } from "./internal/net/StitchAuthRequest";
import StitchEvent from "./internal/net/StitchEvent";
import StitchRequestClient from "./internal/net/StitchRequestClient";
import AuthRebindEvent from "./services/internal/AuthRebindEvent";
import CoreStitchServiceClientImpl from "./services/internal/CoreStitchServiceClientImpl";
import { RebindEvent } from "./services/internal/RebindEvent";
import StitchServiceRoutes from "./services/internal/StitchServiceRoutes";
import { StitchAppClientConfiguration } from "./StitchAppClientConfiguration";
import StitchAppClientInfo from "./StitchAppClientInfo";
import StitchClientError from "./StitchClientError";
import { StitchClientErrorCode } from "./StitchClientErrorCode";
import StitchError from "./StitchError";
import StitchRequestError from "./StitchRequestError";
import { StitchRequestErrorCode } from "./StitchRequestErrorCode";
import StitchServiceError from "./StitchServiceError";
import { StitchServiceErrorCode } from "./StitchServiceErrorCode";
import Stream from "./Stream";
export { BSON, AuthInfo, StitchAuthResponseCredential, AnonymousAuthProvider, AnonymousCredential, ApiStitchUserIdentity, CustomAuthProvider, CustomCredential, FacebookAuthProvider, FacebookCredential, GoogleAuthProvider, GoogleCredential, ServerApiKeyAuthProvider, ServerApiKeyCredential, UserApiKeyAuthProvider, UserApiKey, UserApiKeyCredential, StitchError, StitchClientError, StitchClientErrorCode, CoreUserApiKeyAuthProviderClient, UserPasswordAuthProvider, UserPasswordCredential, CoreUserPassAuthProviderClient, CoreStitchAppClient, CoreStitchAuth, CoreStitchUserImpl, DeviceFields, BasicRequest, ContentTypes, Event, BaseEventStream, StitchEvent, Headers, Stream, StitchAppClientInfo, StitchAppClientConfiguration, StitchAppRequestClient, StitchAppRoutes, StitchAuthRequest, StitchRequestClient, StitchRequestError, StitchRequestErrorCode, StitchServiceRoutes, StitchServiceError, StitchServiceErrorCode, StitchUserProfileImpl, CoreStitchServiceClientImpl, StitchUserIdentity, StitchAppAuthRoutes, Method, Response, MemoryStorage, handleRequestError, UserType, Assertions, AuthEventKind, RebindEvent, AuthRebindEvent, base64Decode, base64Encode, utf8Slice };
